generators/app/templates/src/StarterKit/DataAccess/Repositories/EntityRepositoryBase.cs

```diff
@@ -0,0 +1,273 @@
+    {
+      IQueryable<TEntity> query = Context.Set<TEntity>();
+
+      if (includes != null)
+      {
+        query = includes(query);
+      }
+
+      return query.SingleOrDefaultAsync(x => Equals(x.Id, id));
+    }
+
+    public virtual IEnumerable<TEntity> Query(Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return result.ToList();
+    }
+
+    public virtual async Task<IEnumerable<TEntity>> QueryAsync(Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return await result.ToListAsync();
+    }
+
+    public virtual void Load(Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      result.Load();
+    }
+
+    public virtual async Task LoadAsync(Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      await result.LoadAsync();
+    }
+
+    public virtual IEnumerable<TEntity> QueryPage(int startRow, int pageLength, Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return result.Skip(startRow).Take(pageLength).ToList();
+    }
+
+    public virtual async Task<IEnumerable<TEntity>> QueryPageAsync(int startRow, int pageLength,
+      Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return await result.Skip(startRow).Take(pageLength).ToListAsync();
+    }
+
+    public virtual void Add(TEntity entity)
+    {
+      if (entity == null) throw new InvalidOperationException("Unable to add a null entity to the repository.");
+      Context.Set<TEntity>().Add(entity);
+    }
+
+    public virtual void AddBatch(IEnumerable<TEntity> entities)
+    {
+      if (entities == null) throw new ArgumentNullException(nameof(entities));
+      Context.Set<TEntity>().AddRange(entities);
+    }
+
+    public virtual TEntity Update(TEntity entity)
+    {
+      return Context.Set<TEntity>().Update(entity).Entity;
+    }
+
+    public virtual IEnumerable<TEntity> UpdateBatch(IEnumerable<TEntity> entities)
+    {
+      if (entities == null) throw new ArgumentNullException(nameof(entities));
+      return entities.Select(Update);
```

Comment:
waarom niet updaterange?

Reply:
@RobLiekens Dit had ik niet gedaan omdat de returntype van update range void is. Bij de enkele update werd entity terugestuurd dus ik wilde dit ook graag bij de updateBatch houden om het gelijk te houden.
Indien gewenst kan ik het uiteraard aanpassen naar update range. Dan is alleen de vraag wat je als returntype verwacht. Is void goed of wil je wel de geupdate entities terug?

Reply:
Ik zou gewoon void returnen, er is geen meerwaarde om een entity terug te geven na een update omdat je alle gegevens al hebt. Bij create ligt dit natuurlijk anders omdat er dan een id in het return object zit, maar bij update heeft dit meestal geen nut. Performance is ook slechter als je het op deze manier zou laten staan.

Reply:
Top, helemaal mee eens, pas ik dat aan. Nog wel een ander punt dan want juist bij de create stond een void. Dat had ik zo gelaten maar ik kan mij, net als jou, goed voorstellen dat je het daar juist wel wilt. Wat wil je daar mee?

Reply:
Ja bij create range is dat eigenlijk ook normaal, want je kan niet weten welk object aan welke id gekoppeld is als je een lijst zou terugkrijgen. Mag blijven staan dus